### ISSUE TITLE
Custom desktop entries in home dir should persist if the global variant updates

### ISSUE THREAD

**Issue by friday** [Ulauncher/Ulauncher#320]
**Steps to reproduce:**
1. Copy a desktop entry from `/usr/share/applications/` to `~/.local/share/applications`
2. Add "NoDisplay=true" to the new file.
3. Reinstall/rebuild or upgrade software that recreates the original desktop entry.

**Expected result:**
After step 2, the desktop entry should be excluded from the results, and as long as the custom entry in the home dir remains there it should stay hidden regardless of changes to the original file.

**Result:**
When I upgrade a package with a desktop entry in `/usr/share/applications/`, Ulauncher loads it, even if there's a version in my home dir with the same name. Restarting Ulauncher fixes it.
(I actually had to restart after step 2 as well, but this is a different issue).

It seems the `add_file`-methods in app_watcher.py needs a little more logic to handle this. Didn't add the log since this should happen to everyone.

**Linux distro**: Arch (Antergos)
**Ulauncher version**: 4.3.1.r4-1
**Desktop environment**: Gnome 
**DESKTOP_DIRS**: `['/home/albin/.local/share/applications', '/usr/share/applications']`

**Comment by gornostal** [comment]
Thanks for reporting this bug @friday

**Comment by friday** [comment]
Fixed in #835 (merged into the V6 branch).
